app/components/feedback-button/component.ts

```diff
@@ -0,0 +1,133 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import config from 'ember-get-config';
+import $ from 'jquery';
+
+const {
+    microfeedback: { enabled, pageParams, url },
+    OSF: { backend },
+} = config;
+
+function sendFeedback(body: string, {
+    extra,
+    followup,
+    pageName,
+    userID,
+}: {
+    extra?: object,
+    followup: boolean,
+    pageName: string,
+    userID: string,
+}): Promise<any> {
+    const payload = {
+        body,
+        extra: {
+            followup,
+            user: userID ? `[${userID}|${window.location.origin}/${userID}/]` : undefined,
+            ...extra,
+        },
+    };
+
+    const params = [
+        (backend !== 'prod') ? { name: 'label', value: 'testing' } : undefined,
+        followup ? { name: 'label', value: 'followup' } : undefined,
+        ...Object.entries(pageParams[pageName] || {})
+            .map(([ name, value ]) => ({ name, value })),
+    ].filter((val) => !!val);
+
+    if (!url) {
+        return;
+    }
+
+    return $.ajax(`${url}?${$.param(params)}`, {
+        contentType: 'application/json',
+        data: JSON.stringify(payload),
+        dataType: 'json',
+        method: 'POST',
+    });
+}
+
+enum DialogState {
+    empty = 'empty',
+    active = 'active',
+    success = 'success',
+}
+
+export default class FeedbackButton extends Component.extend({
+    pageName:  null,
```

Comment:
This is an object property, so it's being initialized to null (like in the JS version). The type is inferred from the value.

Reply:
Right, it infers type from the value `null`. I was wrong, it'll actually infer type `any`, not `null`, but it'd be nice to have explicit types instead of `any` wherever we can.